<launch>
    <arg name="robot_name" default="sparkal" />
    <arg name="sim_time_required" default="true"/>
    <param name="use_sim_time" value="$(arg sim_time_required)"/>

    <arg name="dsg_output_dir" default=""/>
    <arg name="dsg_output_prefix" default=""/>

    <arg name="use_semantics" default="true"/>
    <arg name="use_bag_semantics" default="true"/>
    <arg name="use_gt_poses" default="false"/>
    <arg name="sensor_frame" default="left_cam"/>
    <arg name="publish_archived" default="true"/>
    <arg name="throttle_semantics" default="false"/>

    <arg name="use_lidar" default="false"/>

    <arg name="enable_rectification" default="true"/>

    <include file="$(find hydra_utils)/launch/gt_pose/d455_gt_pose.launch" if="$(arg use_gt_poses)">
        <arg name="sensor_frame" value="$(arg sensor_frame)"/>
    </include>

    <arg name="rgb_info_topic" default="/$(arg robot_name)/forward/color/camera_info"/>
    <arg name="rgb_topic" default="/$(arg robot_name)/forward/color/image_raw" unless="$(arg use_bag_semantics)"/>
    <arg name="rgb_topic" default="/$(arg robot_name)/oneformer/image_raw" if="$(arg use_bag_semantics)"/>
    <arg name="depth_topic" default="/$(arg robot_name)/forward/depth/image_rect_raw"/>
    <arg name="depth_info_topic" default="/$(arg robot_name)/forward/depth/camera_info"/>
    <arg name="pointcloud_topic" default="/$(arg robot_name)/semantic_pointcloud" unless="$(arg use_lidar)"/>
    <arg name="pointcloud_topic" default="/$(arg robot_name)/lidar_points" if="$(arg use_lidar)"/>

    <include file="$(find hydra_utils)/launch/includes/rgbd_to_pointcloud.xml" pass_all_args="true"
             if="$(eval arg('use_bag_semantics') or not arg('use_semantics'))"/>

    <include file="$(find semantic_recolor)/launch/semantic_recolor_pointcloud.launch" pass_all_args="true"
        if="$(eval arg('use_semantics') and not arg('use_bag_semantics'))">
        <arg name="model_name" default="hrnetv2_360_640_v12"/>
        <arg name="rgb_image_transport" value="compressed"/>
    </include>

    <include file="$(find hydra_topology)/launch/hydra_topology.launch" pass_all_args="true">
        <arg name="config" value="simmons_topology_config.yaml" unless="$(arg use_lidar)"/>
        <arg name="config" value="simmons_lidar_topology_config.yaml" if="$(arg use_lidar)"/>
        <arg name="semantic_config" value="ade150_semantic_config.yaml"/>
        <arg name="semantic_color_path" value="$(find semantic_recolor)/config/colors/ade150_config.csv"/>
        <arg name="voxel_size" value="0.20" unless="$(arg use_lidar)"/>
        <arg name="voxel_size" value="0.50" if="$(arg use_lidar)"/>
        <arg name="max_ray_length_m" value="3.0" unless="$(arg use_lidar)"/>
        <arg name="max_ray_length_m" value="5.0" if="$(arg use_lidar)"/>
    </include>
</launch>
